biogeochem/FatesAllometryMod.F90

```diff
@@ -2586,6 +2589,182 @@ subroutine ForceDBH( ipft, crowndamage, canopy_trim, elongf_leaf, elongf_stem, d
      return
   end subroutine ForceDBH
 
+  ! =========================================================================
+
+  subroutine VegAreaLayer(tree_lai,tree_sai,tree_height,iv,nv,pft,snow_depth, & 
+       vai_top,vai_bot, elai_layer,esai_layer,tlai_layer,tsai_layer)
+
+    ! -----------------------------------------------------------------------------------
+    ! This routine returns the exposed leaf and stem areas (m2 of leaf and stem) per m2 of
+    ! ground inside the crown, for the leaf-layer specified.
+    ! -----------------------------------------------------------------------------------
+
+    real(r8),intent(in) :: tree_lai         ! the in-crown leaf area index for the plant
+                                            ! [m2 leaf/m2 crown footprint]
+    real(r8),intent(in) :: tree_sai         ! the in-crown stem area index for the plant
+                                            ! [m2 stem/m2 crown footprint]
+    real(r8),intent(in) :: tree_height      ! the height of the plant [m]
+    integer,intent(in)  :: iv               ! vegetation layer index
+    integer,intent(in)  :: nv               ! this plants total number of veg layers
+    integer,intent(in)  :: pft              ! plant functional type index
+    real(r8),intent(in) :: snow_depth       ! the depth of snow on the ground [m]
+    real(r8),intent(out) :: vai_top
+    real(r8),intent(out) :: vai_bot          ! the VAI of the bin top and bottom
+    real(r8),intent(out) :: elai_layer       ! exposed leaf area index of the layer
+    real(r8),intent(out) :: esai_layer       ! exposed stem area index of the layer
+    real(r8),optional,intent(out) :: tlai_layer       ! total leaf area index of the layer
+    real(r8),optional,intent(out) :: tsai_layer       ! total stem area index of the layer
+
+                                 ! [m2 of leaf in bin / m2 crown footprint]
+    real(r8) :: tree_vai         ! the in-crown veg area index for the plant
+    real(r8) :: fraction_exposed ! fraction of the veg media that is above snow
+    real(r8) :: layer_top_height ! Physical height of the layer top relative to ground [m]
+    real(r8) :: layer_bot_height ! Physical height of the layer bottom relative to ground [m]
+    real(r8) :: tlai,tsai        ! temporary total area indices [m2/m2]
+    real(r8) :: fleaf            ! fraction of biomass in layer that is leaf
+    real(r8) :: remainder        ! old-method: remainder of biomass in last bin
+    integer, parameter :: layer_height_const_depth = 1 ! constant physical depth assumption
+    integer, parameter :: layer_height_const_lad   = 2 ! constant leaf area depth assumption
+    integer, parameter :: layer_height_method = layer_height_const_depth
```

Comment:
The constant depth assumption is the default, and is equivalent to the way we currently do things in the canopy summarization code.  I added the alternative so that we have the option to explore it in the future. I have not done a thorough investigation of it yet though.